src/layers/singleLayer.jl

```diff
@@ -178,7 +177,7 @@ function JthetaTmv(this::singleLayer{T},Zin::Array{T},dummy::Array{T},theta::Arr
     Yout,dummy,tmpNL  = apply(this.nLayer,th4,Yout,[])
     Yout .+= this.Bin * th2
 	A,dA   = this.activation(Yout,true)
```

Comment:
Can this activation not be done in place? something like this.activation!(Yout, similar(Yout), true)